doc/nrf/libraries/zigbee/lib_zigbee_signal_handler.rst

```diff
@@ -0,0 +1,296 @@
+﻿.. _lib_zigbee_signal_handler:
+
+Zigbee default signal handler
+#############################
+
+The :ref:`nrfxlib:zboss` interacts with the user application by invoking the :cpp:func:`zboss_signal_handler()` function whenever a stack event, such as network steering, occurs.
+It is mandatory to define :cpp:func:`zboss_signal_handler()` in the application.
+
+Because most of Zigbee devices behave in a similar way, :cpp:func:`zigbee_default_signal_handler()` was introduced to provide a default logic to handle stack signals.
+
+.. note::
+    |zigbee_library|
+
+.. _zarco_signal_handler_minimal:
+
+Minimal zboss_signal_handler implementation
+*******************************************
+
+This function can be called in the application's :cpp:func:`zboss_signal_handler()` to simplify the implementation.
+In such case, this minimal implementation includes only a call to the default signal handler:
+
+
+.. code-block:: c
+
+    void zboss_signal_handler(zb_uint8_t param)
+    {
+        /* Call default signal handler. */
+        zigbee_default_signal_handler(param);
+
+        /* Free the Zigbee stack buffer. */
+        zb_buf_free(param);
+    }
+
+With this call, the device will be able to join the Zigbee network or join a new network when it leaves the previous one.
+
+In general, using the default signal handler is worth considering because of the following reasons:
+
+* It simplifies the application.
+* It provides a default behavior for each signal, which reduces the risk that an application will break due to an unimplemented signal handler.
+* It provides Zigbee role specific behavior (for example, finite join/rejoin time)
+* It makes the application less sensitive to changes in the Zigbee stack commissioning API.
+
+Apart from that, the default signal handler serves as a good starting point for a custom signal handler implementation.
+
+.. _zarco_signal_handler_full:
+
+Complete zboss_signal_handler implementation
+********************************************
+
+In its complete implementation, the `zboss_signal_handler` allows the application to control a broader set of basic functionalities, including joining, commissioning, and network formation.
+
+There are in fact cases in which the default handler will not be sufficient and needs to be extended.
+For example, when the application wants to use the procedure of the initiator of finding & binding or use the production configuration feature.
+
+If the application wants to change or extend the default behavior, use a switch-case statement and then the default signal handler inside the default case:
+
+.. code-block:: c
+
+    void zboss_signal_handler(zb_bufid_t bufid)
+    {
+        zb_zdo_app_signal_hdr_t  * p_sg_p = NULL;
+        zb_zdo_app_signal_type_t   sig    = zb_get_app_signal(bufid, &p_sg_p);
+        zb_ret_t                   status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+
+        switch (sig)
+        {
+            case ZB_BDB_SIGNAL_DEVICE_REBOOT:
```

Comment:
I would add here also `ZB_BDB_SIGNAL_DEVICE_FIRST_START` because `start_network_rejoin()` is used for this signal in the `default_signal_handler()`.
Similarly, `ZB_ZDO_SIGNAL_LEAVE` should be added here. If default behavior is to be replaced, it should be replaced completely.

Reply:
What do you think about replacing `bdb_start_top_level_commissioning` with `start/stop_network_rejoin()`?

Reply:
It depends on what is the purpose of this section. Do we want to have implementation which does not use `start/stop_network_rejoin()` or use the same implementation as `default_signal_handler()`?

Reply:
It is already blended in default signal handler, so I do not see much harm in calling it here. Additionally - there is no Kconfig switch to disable the rejoin process, so it has to be managed by the app if it does not override all signal handlers.

Reply:
This section is now splitted to two subsections and only `bdb_start_top_level_commissioning()` used in doc snippets.